lib/ts/recipe/webauthn/recipeImplementation.ts

```diff
@@ -0,0 +1,514 @@
+                }
+
+                if (
+                    error.name === "NotSupportedError" ||
+                    error.message === "WebAuthn is not supported in this browser"
+                ) {
+                    return { status: "WEBAUTHN_NOT_SUPPORTED", error: error };
+                }
+
+                return {
+                    status: "FAILED_TO_REGISTER_USER",
+                    error: error,
+                };
+            }
+
+            return {
+                status: "OK",
+                registrationResponse,
+            };
+        },
+        registerCredentialWithSignUp: async function ({ email, options, userContext }) {
+            // Get the registration options by using the passed email ID.
+            const registrationOptions = await this.getRegisterOptions({ options, userContext, email });
+            if (registrationOptions?.status !== "OK") {
+                // If we did not get an OK status, we need to return the error as is.
+
+                // If the `status` is `RECOVER_ACCOUNT_TOKEN_INVALID_ERROR`, we need to throw an
+                // error since that should never happen as we are registering with an email
+                // and not a token.
+                if (registrationOptions?.status === "RECOVER_ACCOUNT_TOKEN_INVALID_ERROR") {
+                    throw new Error("Got `RECOVER_ACCOUNT_TOKEN_INVALID_ERROR` status that should never happen");
+                }
+
+                return registrationOptions;
+            }
+
+            // We should have received a valid registration options response.
+            const registerCredentialResponse = await this.registerCredential({ registrationOptions });
+            if (registerCredentialResponse.status !== "OK") {
+                return registerCredentialResponse;
+            }
+
+            // We should have a valid registration response for the passed credentials
+            // and we are good to go ahead and verify them.
+            return await this.signUp({
+                webauthnGeneratedOptionsId: registrationOptions.webauthnGeneratedOptionsId,
+                credential: registerCredentialResponse.registrationResponse,
+                options,
+                userContext,
+            });
+        },
+        authenticateCredential: async function ({ authenticationOptions }) {
+            let authenticationResponse: AuthenticationResponseJSON;
+            try {
+                authenticationResponse = await startAuthentication({ optionsJSON: authenticationOptions });
+            } catch (error: any) {
+                if (
+                    error.name === "NotSupportedError" ||
+                    error.message === "WebAuthn is not supported in this browser"
+                ) {
+                    return { status: "WEBAUTHN_NOT_SUPPORTED", error: error };
+                }
+
+                return {
+                    status: "FAILED_TO_AUTHENTICATE_USER",
```

Comment:
When does this happen? The status code feels a bit misleading (Seeing this I'd assume that for some reason the backend rejected my login attempt)